floodmodeller_api/test/test_toolbox_structure_log.py

```diff
@@ -79,156 +83,110 @@ def structure():
-    assert output == ["Mannings: 0", "dia: 0.00 x l: 0.00"]
-    output = slb._circular_data(conduit_filled, with_length)
-    assert output == [
-        "Mannings: [min: 1.453345, max: 3.435]",
-        "dia: 6.00 x l: 4.97",
-    ]
-
-
-def test_sprungarch_data(slb, conduit_empty, conduit_filled, no_length, with_length):
-    output = slb._sprungarch_data(conduit_empty, no_length)
-    assert output == [
-        "Mannings: 0",
-        "(Springing: 0.00, Crown: 0.00) x w: 0.00 x l: 0.00",
-    ]
-    output = slb._sprungarch_data(conduit_filled, with_length)
-    assert output == [
-        "Mannings: [min: 1.34, max: 1.876]",
-        "(Springing: 23.10, Crown: 5.40) x w: 3.00 x l: 4.97",
-    ]
-
-
-def test_rectangular_data(slb, conduit_empty, conduit_filled, no_length, with_length):
-    output = slb._rectangular_data(conduit_empty, no_length)
-    assert output == ["Mannings: 0", "h: 0.00 x w: 0.00 x l: 0.00"]
-    output = slb._rectangular_data(conduit_filled, with_length)
-    assert output == [
-        "Mannings: [min: 1.34, max: 1.876]",
-        "h: 25.45 x w: 3.00 x l: 4.97",
-    ]
-
-
-def test_section_data(slb, conduit_empty, conduit_filled, no_length, with_length):
-    output = slb._section_data(conduit_empty, no_length)
-    assert output == ["Colebrook-White: 0", "h: 0.00 x w: 0.00 x l: 0.00"]
-    output = slb._section_data(conduit_filled, with_length)
-    assert output == [
-        "Colebrook-White: [min: 0.0, max: 4.0]",
-        "h: 65.00 x w: 150.00 x l: 4.97",
-    ]
-
-
-def test_sprung_data(slb, conduit_empty, conduit_filled, no_length, with_length):
-    output = slb._sprung_data(conduit_empty, no_length)
-    assert output == [
-        "Mannings: 0",
-        "(Springing: 0.00, Crown: 0.00) x w: 0.00 x l: 0.00",
-    ]
-    output = slb._sprung_data(conduit_filled, with_length)
-    assert output == [
-        "Mannings: [min: 1.34, max: 1.876]",
-        "(Springing: 23.10, Crown: 5.40) x w: 3.00 x l: 4.97",
-    ]
-
-
-def test_orifice_dimensions(slb, structure):
-    structure.invert = 1
-    output = slb._orifice_dimensions(structure)
-    assert output == "h: -1.00 x w: -1.00"
-
-
-def test_spill_data(slb, structure):
-    structure.data = pd.DataFrame(data={"X": [0, 0], "Y": [0, 0]})
-    structure.weir_coefficient = 0
-    output = slb._spill_data(structure)
-    assert output == ["Elevation: 0.00 x w: 0.00", 0]
-
-
-def test_bridge_data(slb, structure):
-    structure.section_data = pd.DataFrame(data={"X": [0, 0], "Y": [0, 0], "Mannings n": [0, 0]})
-    structure.opening_data = pd.DataFrame(
-        data={"Start": 0, "Finish": 0, "Springing Level": 0, "Soffit Level": 0},
-        index=[0],
-    )
-    output = slb._bridge_data(structure)
-    assert output == ["Mannings: 0", "h: 0.00 x w: 0.00"]
+@pytest.fixture
+def ex18_dat_expected():
+    # I think this is the correct way to establish the same 'expected' result, but if this is a larger/more structure-rich dat, could read from file?
```

Comment:
I think this is okay for such a small dat, but if it was much bigger, then I would read it from a file